src/Network.cpp

```diff
@@ -233,36 +233,27 @@ std::pair<int, int> Network::load_v1_network(std::istream& wtfile) {
                 process_bn_var(weights);
                 batchnorm_stddivs.emplace_back(weights);
             }
-        } else if (linecount == plain_conv_wts) {
-            conv_pol_w = std::move(weights);
-        } else if (linecount == plain_conv_wts + 1) {
-            conv_pol_b = std::move(weights);
-        } else if (linecount == plain_conv_wts + 2) {
-            std::copy(cbegin(weights), cend(weights), begin(bn_pol_w1));
-        } else if (linecount == plain_conv_wts + 3) {
-            process_bn_var(weights);
-            std::copy(cbegin(weights), cend(weights), begin(bn_pol_w2));
-        } else if (linecount == plain_conv_wts + 4) {
-            std::copy(cbegin(weights), cend(weights), begin(ip_pol_w));
-        } else if (linecount == plain_conv_wts + 5) {
-            std::copy(cbegin(weights), cend(weights), begin(ip_pol_b));
-        } else if (linecount == plain_conv_wts + 6) {
-            conv_val_w = std::move(weights);
-        } else if (linecount == plain_conv_wts + 7) {
-            conv_val_b = std::move(weights);
-        } else if (linecount == plain_conv_wts + 8) {
-            std::copy(cbegin(weights), cend(weights), begin(bn_val_w1));
-        } else if (linecount == plain_conv_wts + 9) {
-            process_bn_var(weights);
-            std::copy(cbegin(weights), cend(weights), begin(bn_val_w2));
-        } else if (linecount == plain_conv_wts + 10) {
-            std::copy(cbegin(weights), cend(weights), begin(ip1_val_w));
-        } else if (linecount == plain_conv_wts + 11) {
-            std::copy(cbegin(weights), cend(weights), begin(ip1_val_b));
-        } else if (linecount == plain_conv_wts + 12) {
-            std::copy(cbegin(weights), cend(weights), begin(ip2_val_w));
-        } else if (linecount == plain_conv_wts + 13) {
-            std::copy(cbegin(weights), cend(weights), begin(ip2_val_b));
+        } else {
+            const auto offset = linecount - plain_conv_wts;
+            if (offset == 3 || offset == 9) {
+                process_bn_var(weights);
```

Comment:
If process_bn_var is in-place this can just be moved out of the loop and applied to bn_pol_2 and bn_val_v2.